weechatScripts.wee-slack: 2.4.0 -> 2.5.0

bdesham opened the pull request:
###### Motivation for this change

Version bump. The changes are listed on [the GitHub release page](https://github.com/wee-slack/wee-slack/releases/tag/v2.5.0).

wee-slack now (optionally) depends on a data file called weemoji.json
that allows it to translate between Slack emoji names (like
"slight_smile") and Unicode codepoints. For convenience, the derivation
now installs this file and patches the script so that the user doesn't
need to do any extra configuration to use it.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @WilliButz

JeffLabonte reviewed (approved):
LGTM

Mic92 commented on pkgs/applications/networking/irc/weechat/scripts/wee-slack/default.nix:
```diff
@@ -21,11 +21,18 @@ stdenv.mkDerivation rec {
   ];
 
+  postPatch = ''
+    substituteInPlace wee_slack.py --replace \
```
Could you make this a patch instead? This way we will not silently break it, when upgrading wee slack in future. Thanks!

bdesham replied:
Good idea! I’ve updated the PR. I believe it’s still necessary to use `substituteInPlace`, based on what I see being done in other derivations, but at least now the entire build will fail if the patch doesn’t apply cleanly.

bdesham committed:
weechatScripts.wee-slack: 2.4.0 -> 2.5.0

wee-slack now (optionally) depends on a data file called weemoji.json
that allows it to translate between Slack emoji names (like
"slight_smile") and Unicode codepoints. For convenience, the derivation
now installs this file and patches the script so that the user doesn't
need to do any extra configuration to use it.
